### PR TITLE
bpo-11822: Improve disassembly to show embedded code objects.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Lib/dis.py [hunk 1]
```diff
@@ -327,12 +327,18 @@ def _get_instructions_bytes(code, varnames=None, names=None, constants=None,
...
     cell_names = co.co_cellvars + co.co_freevars
     linestarts = dict(findlinestarts(co))
     _disassemble_bytes(co.co_code, lasti, co.co_varnames, co.co_names,
                        co.co_consts, cell_names, linestarts, file=file)
+    if recursive:
+        for x in co.co_consts:
+            if hasattr(x, 'co_code'):
+                print(file=file)
+                print("Disassembly of %r:" % (x,), file=file)
+                disassemble(x, file=file, recursive=True)
```
Comment (ncoghlan) [inline Lib/dis.py:345]: This has the potential to get very noisy if disassembling a whole module, so I've posted a design question back on the original issue: https://bugs.python.org/issue11822#msg291718

I'd prefer to keep that discussion on the issue tracker, so I won't duplicate it here.

**Review by ncoghlan — CHANGES_REQUESTED** [review]
Thanks for getting this RFE moving again!

I think there may be some usability and robustness problems with a simple all-or-nothing flag for the recursive behaviour, so I've posted that question back on the original issue: https://bugs.python.org/issue11822#msg291718

**Comment by ncoghlan** [comment]
As per the discussion at https://bugs.python.org/issue11822#msg291780, I still prefer the approach with the `depth` parameter, but amended so the default is `depth=None` to automatically render the entire tree.

Link to Serhiy's PR with the `depth` parameter instead of the `recursive` flag: https://github.com/python/cpython/pull/1844

**Comment by serhiy-storchaka** [comment]
Closed in favor of #1844.

**Pull request closed without merging**